rust_src/src/threads.rs

```diff
@@ -53,4 +62,21 @@ pub fn thread_alive_p(thread: ThreadStateRef) -> bool {
     thread.is_alive()
 }
 
+/// Return the current thread.
+#[lisp_fn]
+pub fn current_thread() -> LispObject {
+    ThreadState::current_thread().as_lisp_obj()
+}
```

Comment:
You can make the `lisp_fn` macro do more of the work by adding a `From` impl:

```
impl From<ThreadStateRef> for LispObject {
    pub fn from(th: ThreadStateRef) -> LispObject {
        th.as_lisp_obj()
    }
}
```

However, this can be done later; I won't hold this patch up because of it.